### PR TITLE
cleanup(mixin): add one API test case for location mixin

### PR DESCRIPTION
**Pull request by cuiy0006**
In this PR, I add one API test case(`Locations` - `GetLocation`) for review.

After it is merged, I will add one API test case for each of LRO and IAM.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14813)

### CODE REVIEW

File: generator/integration_tests/tests/golden_thing_admin_stub_test.cc [hunk 1]
```diff
@@ -1002,6 +1077,7 @@ TEST_F(GoldenStubTest, AsyncCancelOperation) {
   CompletionQueue cq(mock);
 
   DefaultGoldenThingAdminStub stub(std::move(grpc_stub_),
+                                   std::move(location_stub_),
                                    std::move(longrunning_stub_));
   google::longrunning::CancelOperationRequest request;
```
[the remark this answers is not in the document]
Comment (cuiy0006) [reply generator/integration_tests/tests/golden_thing_admin_stub_test.cc:1099]: Done.

File: generator/integration_tests/tests/golden_thing_admin_stub_test.cc [hunk 2]
```diff
@@ -543,11 +543,61 @@ class MockLongrunningOperationsStub
               (override));
 };
 
+class MockLocationOperationsStub
```
[the remark this answers is not in the document]
Comment (cuiy0006) [reply generator/integration_tests/tests/golden_thing_admin_stub_test.cc]: Done

File: generator/integration_tests/tests/golden_kitchen_sink_stub_test.cc [hunk 3]
```diff
@@ -350,10 +350,60 @@ class MockGrpcGoldenKitchenSinkStub : public ::google::test::admin::database::
       (override));
 };
 
+class MockLocationOperationsStub
```
[the remark this answers is not in the document]
Comment (cuiy0006) [reply generator/integration_tests/tests/golden_kitchen_sink_stub_test.cc:353]: Done

**Review by dbolduc — APPROVED** [review]
Thanks

**Pull request merged**